### ISSUE TITLE
Method craft\\elements\\ElementCollection::clearCachedResult does not exist.

### ISSUE THREAD

**Issue by Anubarak** [engram-design/SuperTable#513]
### Describe the bug

https://github.com/verbb/super-table/blob/craft-4/src/fields/SuperTableField.php#L1118

could you please move the line `$value->clearCachedResult();` inside the instanceof condition
```
if ($value instanceof SuperTableBlockQuery) {
    $this->_populateQuery($value, $element);
    $value->clearCachedResult();
}
```
`$value` can be instance of `ElementCollection` or `SuperTableBlockQuery` and only the Query has this function.

### Steps to reproduce

1. eager load a supertable field value
2. save the owner

### Craft CMS version

4.X

### Plugin version

3.0.7

### Multi-site?

"No"

**Comment by engram-design** [comment]
Fixed for the next release. To get this early, run `composer require verbb/super-table:"dev-craft-4 as 3.0.7"`.

**Comment by engram-design** [comment]
Fixed in [3.0.8](https://github.com/verbb/super-table/releases/tag/3.0.8)
